### PR TITLE
Homework Podstawy C++ #2

### PR DESCRIPTION
**Pull request by MechaniQ1**
@MechaniQ1 @MateuszKwiecien 

All tasks done.

### CODE REVIEW

File: module2/homework/addEven/addEven.hpp [hunk 1]
```diff
@@ -2,7 +2,10 @@
 #include <vector>
 
 int addEven(const std::vector<int>& numbers) {
-    // TODO: Your implementation goes here
-    // Below return is only to make this function compile now
-    return -1;
+    int sum = 0;
+    for (auto i = 0; i < numbers.size(); i++)
+    {
+        if (numbers[i]%2 == 0) sum += numbers[i];
```
Comment (nauka-programowania-MA) [inline module2/homework/addEven/addEven.hpp:8]: space between %

File: module2/homework/generateSequence/generateSequence.hpp [hunk 2]
```diff
@@ -2,6 +2,13 @@
 #include <vector>
 
 std::vector<int> generateSequence(int count, int step) {
-    // TODO: Implement me :)
-    return {};
+    if(count<=0){
```
Comment (nauka-programowania-MA) [inline module2/homework/generateSequence/generateSequence.hpp:5]: space

File: module2/homework/generateSequence/generateSequence.hpp [hunk 3]
```diff
@@ -2,6 +2,13 @@
...
+        return {};
+    }
+    std::vector<int> sequence;
+    sequence.reserve(count);
+    for(size_t i = 0; i<count; i++){
```
Comment (nauka-programowania-MA) [inline module2/homework/generateSequence/generateSequence.hpp:10]: space

File: module2/homework/generateSequence/generateSequence.hpp [hunk 4]
```diff
@@ -2,6 +2,13 @@
...
+        sequence.push_back(step*(i+1));
```
Comment (nauka-programowania-MA) [inline module2/homework/generateSequence/generateSequence.hpp:11]: space

File: module2/homework/nwdNww/nwdNww.hpp [hunk 5]
```diff
@@ -1,11 +1,33 @@
...
+        return lhs;
+    }
+    if(lhs<0){
+        lhs=-1*lhs;
+    }
+    if(rhs<0){
+        rhs=-1*rhs;
+    }
+    int temp;
+    while(rhs!=0){
```
Comment (nauka-programowania-MA) [inline module2/homework/nwdNww/nwdNww.hpp:14]: spaces

File: module2/homework/nwdNww/nwdNww.hpp [hunk 6]
```diff
@@ -1,11 +1,33 @@
...
-    return -1;
+    if(lhs==0||rhs==0){
+        return 0;
+    }
+    if(lhs<0){
+        lhs=-1*lhs;
+    }
+    if(rhs<0){
+        rhs=-1*rhs;
+    }
```
Comment (nauka-programowania-MA) [inline module2/homework/nwdNww/nwdNww.hpp:31]: spaces

**Comment by ziobron** [comment]
+4 XP addEven
+6 XP NwwNwd
+5 XP maxOfVector
+5 XP generateSequence
+8 XP delivery before 31.05
+3 XP cooperation

{
  "requirement_ids": [1119, 1631, 1119, 1633, 1119, 1634, 1119, 1635, 1636],
  "points": [2, 4, 2, 6, 2, 5, 2, 5, 3],
  "users": ["MechaniQ1", "MateuszKwiecien"]
}

**Pull request closed without merging**